src/knot_cloud.c

```diff
@@ -117,110 +96,77 @@ static int map_routing_key_to_msg_type(const char *routing_key)
 }
 
 static struct knot_cloud_msg *create_msg(const char *routing_key,
-					 json_object *jso)
+					 const char *json_str)
 {
+	bool has_err;
+
 	struct knot_cloud_msg *msg = l_new(struct knot_cloud_msg, 1);
 
 	msg->type = map_routing_key_to_msg_type(routing_key);
 
-	switch (msg->type) {
-	case UPDATE_MSG:
+	has_err = false;
+	if (msg->type == LIST_MSG) {
+		msg->device_id = NULL;
+	} else {
+		msg->device_id = parser_get_key_str_from_json_str(json_str,
+			KNOT_JSON_FIELD_DEVICE_ID);
+		has_err = msg->device_id ? false : true;
+	}
+
+	if (msg->type == UPDATE_MSG || msg->type == REQUEST_MSG) {
```

Comment:
this if can be replaced by validating if KNOT_JSON_FIELD_ERROR exists in json_str. If does, validates if field is str or null, otherwise sets error=NULL